### PR TITLE
Routing refactoring

### PR DESCRIPTION
**Pull request by Aleks171**

### CODE REVIEW

File: routes/categoriesRoutes.js [hunk 1]
```diff
@@ -1,28 +1,22 @@
...
+  addCategory,
+  getCategories,
 } = require('../models/categories/categoriesAPI');
 const categoriesRoutes = router => {
   router
     .route('/categories')
     .get((req, res) => {
-      getAllCategories().then(category => {
-        res.json(category);
+      getCategories(req.query).then(categories => {
```
Comment (demns) [inline routes/categoriesRoutes.js]: no validation??????

File: server.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ connectToDatabase();
 app.use(
   bodyParser.urlencoded({
     extended: true,
-  }),
```
Comment (demns) [inline server.js]: why

File: server.js [hunk 3]
```diff
@@ -19,8 +19,9 @@ app.use((req, res, next) => {
   res.header('Access-Control-Allow-Origin', '*');
   res.header(
     'Access-Control-Allow-Headers',
-    'Origin, X-Requested-With, Content-Type, Accept',
+    'Origin, X-Requested-With, Content-Type, Accept'
   );
+  res.header('Access-Control-Allow-Methods', 'GET,PUT,POST,DELETE');
```
Comment (demns) [inline server.js]: what is not allowed?

Comment (Aleks171) [reply server.js]: patch

File: models/libraryItems/validateLibraryQuery.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+const validate = value => /^[A-z]+$/.test(value);
```
Comment (demns) [inline models/libraryItems/validateLibraryQuery.js]: https://stackoverflow.com/a/4923397

File: models/news/ObjectIdCheck.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const mongoose = require('mongoose');
+
+const validateObjectId = id => mongoose.Types.ObjectId.isValid(id);
+
+module.exports = validateObjectId;
```
Comment (demns) [inline models/news/ObjectIdCheck.js]: start a function name with a verb, then tell what it does.

File: models/libraryItems/validateLibraryQuery.js (hunk 4 above)
Comment (Aleks171) [reply models/libraryItems/validateLibraryQuery.js]: ok, made better

File: models/news/ObjectIdCheck.js (hunk 5 above)
Comment (Aleks171) [reply models/news/ObjectIdCheck.js]: done

File: models/libraryItems/libraryItemAPI.js [hunk 6]
```diff
@@ -1,18 +1,21 @@
...
   const itemToAdd = new libraryItem(item);
   return itemToAdd.save();
 };
-const getAllItems = () => libraryItem.find({});
 
-const searchByType = typeName => libraryItem.find({ type: typeName });
-
-const searchItemsInCategory = searchQuiery => libraryItem.find(searchQuiery);
+const getItems = searchQuiery => {
+  const { categoryTag, type } = searchQuiery;
```
Comment (demns) [inline models/libraryItems/libraryItemAPI.js]: query

Comment (Aleks171) [reply models/libraryItems/libraryItemAPI.js]: yeah

**Pull request merged**